wagtail/core/management/commands/create_log_entries_from_revisions.py

```diff
@@ -0,0 +1,49 @@
+from django.core.management.base import BaseCommand
+from wagtail.core.models import PageLogEntry, PageRevision
+
+
+def get_comparison(page, revision_a, revision_b):
+    comparison = page.get_edit_handler().get_comparison()
+    comparison = [comp(revision_a, revision_b) for comp in comparison]
+    comparison = [comp for comp in comparison if comp.has_changed()]
+
+    return comparison
+
+
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+        current_page_id = None
+        previous_revision = None
+        for revision in PageRevision.objects.order_by('page_id', 'created_at').select_related('page').iterator():
+            is_new_page = revision.page_id != current_page_id
+            has_content_changes = False
+            current_page_id = revision.page_id
+
+            current_revision_as_page = revision.as_page_object()
+
+            if not PageLogEntry.objects.filter(revision=revision).exists():
+                published = revision.id == revision.page.live_revision_id
+
+                if previous_revision is not None:
+                    comparison = get_comparison(revision.page, current_revision_as_page, previous_revision.as_page_object())
+                    has_content_changes = len(comparison) > 0
+
+                if is_new_page or has_content_changes or published:
+                    if is_new_page:
+                        action = 'wagtail.create'
+                    elif published:
+                        action = 'wagtail.publish'
```

Comment:
I think we should treat publish as a separate event like it is in the view. So if the page was created and published, you have two log entries.